radiasoft/beamsim/codes/elegant.sh

```diff
@@ -1,3 +1,90 @@
+      b. Copies in binary form must include the copyright notice and this
+         Software License Agreement in the documentation and/or other
+         materials provided with the copy.
+
+ 4. You may modify a copy or copies of the Software or any portion of it,
+    thus forming a work based on the Software, and distribute copies of
+    such work outside your organization, if you meet all of the following
+    conditions:
+      a. Copies in source code must include the copyright notice and this
+         Software License Agreement;
+      b. Copies in binary form must include the copyright notice and this
+         Software License Agreement in the documentation and/or other
+         materials provided with the copy;
+      c. Modified copies and works based on the Software must carry
+         prominent notices stating that you changed specified portions of
+         the Software.
+
+ 5. Portions of the Software resulted from work developed under a U.S.
+    Government contract and are subject to the following license: the
+    Government is granted for itself and others acting on its behalf a
+    paid-up, nonexclusive, irrevocable worldwide license in this computer
+    software to reproduce, prepare derivative works, and perform publicly
+    and display publicly.
+
+ 6. WARRANTY DISCLAIMER. THE SOFTWARE IS SUPPLIED "AS IS" WITHOUT
+    WARRANTY OF ANY KIND. THE COPYRIGHT HOLDERS, THEIR THIRD PARTY
+    LICENSORS, THE UNITED STATES, THE UNITED STATES DEPARTMENT OF
+    ENERGY, AND THEIR EMPLOYEES: (1) DISCLAIM ANY WARRANTIES,
+    EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO ANY IMPLIED
+    WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE,
+    TITLE OR NON-INFRINGEMENT, (2) DO NOT ASSUME ANY LEGAL LIABILITY
+    OR RESPONSIBILITY FOR THE ACCURACY, COMPLETENESS, OR USEFULNESS
+    OF THE SOFTWARE, (3) DO NOT REPRESENT THAT USE OF THE SOFTWARE
+    WOULD NOT INFRINGE PRIVATELY OWNED RIGHTS, (4) DO NOT WARRANT
+    THAT THE SOFTWARE WILL FUNCTION UNINTERRUPTED, THAT IT IS
+    ERROR-FREE OR THAT ANY ERRORS WILL BE CORRECTED.
+
+ 7. LIMITATION OF LIABILITY. IN NO EVENT WILL THE COPYRIGHT HOLDERS,
+    THEIR THIRD PARTY LICENSORS, THE UNITED STATES, THE UNITED
+    STATES DEPARTMENT OF ENERGY, OR THEIR EMPLOYEES: BE LIABLE FOR
+    ANY INDIRECT, INCIDENTAL, CONSEQUENTIAL, SPECIAL OR PUNITIVE
+    DAMAGES OF ANY KIND OR NATURE, INCLUDING BUT NOT LIMITED TO LOSS
+    OF PROFITS OR LOSS OF DATA, FOR ANY REASON WHATSOEVER, WHETHER
+    SUCH LIABILITY IS ASSERTED ON THE BASIS OF CONTRACT, TORT
+    (INCLUDING NEGLIGENCE OR STRICT LIABILITY), OR OTHERWISE, EVEN
+    IF ANY OF SAID PARTIES HAS BEEN WARNED OF THE POSSIBILITY OF
+    SUCH LOSS OR DAMAGES.
+EOF
+
+sudo bash << EOF
+mkdir -p $ELEGANT_DOCS
+mv $TMP_LICENSE $ELEGANT_LICENSE
+chown -R root: $ELEGANT_LICENSE
+chmod -v ugo+rx $ELEGANT_DOCS
+chmod -v ugo+r $ELEGANT_LICENSE
+EOF
+
+unset ELEGANT_DOCS
+unset ELEGANT_LICENSE
+unset TMP_LICENSE
+
```

Comment:
We use [delete-trailing-whitespace](https://www.gnu.org/software/emacs/manual/html_node/emacs/Useless-Whitespace.html) in emacs to avoid extra blank lines or spaces at end of lines. While it may seem trivial, it removes a certain amount of noise to not have these extra blanks. I think vim has something like this.